### DISCUSSION TITLE
Can we modify save functionality and linking tool tip (popup) without modifying the existing code?

### DISCUSSION THREAD

**Discussion by sachithsujeewa** [jgraph/drawio#4787] [General]
In the past few months, we have carried out a POC where customers can modify their designs and save them as XML in our file storage. Same time we need to give custom link to elements (which is not a public link) .We've hosted draw.io our own instance for this matter. I understood that now compilation code has been removed from the source code and is there a way we can go forward with this.

**Comment by davidjgraph** [comment]
You still have the source from the version of the repo you cloned.

**Reply by sachithsujeewa** [reply]
We managed to achieve functionality using a custom plugin. is it safe to use a custom plugin approach since we don't want to miss any security patches or updates in future?
